ext/semian/simple_integer.c

```diff
@@ -66,35 +69,68 @@ VALUE semian_simple_integer_increment(int argc, VALUE *argv, VALUE self) {
   VALUE val;
   rb_scan_args(argc, argv, "01", &val);
 
-  semian_simple_integer_shared_t *data = get_value(self);
+  semian_simple_integer_t *res;
+  TypedData_Get_Struct(self, semian_simple_integer_t, &semian_simple_integer_type, res);
+
+  sem_meta_lock(res->sem_id);
```

Comment:
It might be helpful visually to create a new scope here. E.g: 

```
sem_meta_lock(res->sem_id);
{
  // Do work in here
}
sem_meta_unlock();
```

It doesn't change anything functionally, but it makes it clear what the critical section is.
The challenge is that you need to pre-declare variables.

Reply:
TIL!